### ISSUE TITLE
`vue/require-toggle-inside-transition` false negative with whitespace before the element

### ISSUE THREAD

**Issue by yoshi-pi** [vuejs/eslint-plugin-vue#2290]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** ^8.39.0
- **eslint-plugin-vue version:** ^9.11.0
- **Node version:** v18.15.0
- **Operating System:** macOS 13.5.2

**Please show your full configuration:**

```json5
/* eslint-env node */
module.exports = {
  root: true,
  extends: ["plugin:vue/vue3-essential", "eslint:recommended"],
  parserOptions: {
    ecmaVersion: "latest",
  },
};
```

**What did you do?**	

```vue
<template>
  <Transition> <div>Hello</div></Transition>
</template>
```

**What did you expect to happen?**
It should show a red wavy warning of [vue/require-toggle-inside-transition](https://eslint.vuejs.org/rules/require-toggle-inside-transition.html) on the VSCode because there is no v-if of v-show.

**What actually happened?**

It doesn't show any red wavy warnings if there are spaces or line breaks before the element.

**Repository to reproduce this issue**

https://github.com/yoshi-pi/bug-report-vue-require-toggle-inside-transition

**Comment by FloEdelmann** [comment]
Sounds like a bug, thanks for reporting!

**Comment by yoshi-pi** [comment]
Thank you for the changes made. I have three concerns:
1. I noticed that custom components are not getting an error. Why is that? [The <Transition> component](https://vuejs.org/guide/built-ins/transition.html#the-transition-component) can be used with v-if, v-show, \<component\>, and the key attribute. I think it would be a good idea to issue an error if these directives are missing in custom components.
```
<Transition>
  <Foo /> // no error
</Transition>
```
2. Similarly, I think we shouldn't be getting an error for the key attribute for the same reason. I understand that the key attribute is not commonly used, so it's okay to issue an error, but I'm not sure if it's necessary to make an error about it.
```
<Transition>
  <div key="hello" /> // error
</Transition>
```
3. Lastly, when there are multiple elements inside a transition, for example,
```
<Transition>
  <div v-if="ok"></div>
  <div></div>  // no error
</Transition>
```
no error is issued. While this may not be a significant issue because Vue.js shows an overlay in such cases, it still raised a slight concern for me.
What are your thoughts on these three points?

**Comment by 719media** [comment]
I started getting a lot of errors as I use the `key=` syntax noted above in example 2.
I also get errors for
```
<transition>
  <slot />
</transition>
```
Which, IMO, should not be an error
